test/SlowTests/Issues/RavenDB-22295.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Tests.Infrastructure;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues;
+internal class RavenDB_22295 : ClusterTestBase
+{
+    public RavenDB_22295(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    [RavenFact(RavenTestCategory.Cluster)]
+    public async Task UnobservedObjectDisposedExceptionAfterLeaderDispose()
+    {
+        Exception e = null;
+        EventHandler<UnobservedTaskExceptionEventArgs> handler = (sender, args) =>
+        {
+            if(e is ObjectDisposedException)
+                e = args.Exception;
+        };
+
+        TaskScheduler.UnobservedTaskException += handler;
```

Comment:
will this catch exceptions from other tests? is that okay?

Reply:
Done